packages/help-center/src/hooks/use-context-based-search-mapping.tsx

```diff
@@ -10,6 +11,7 @@ interface BlockStore {
 	getBlockType: ( arg0: string ) => { title: string; name: string };
 }
 
+// When using a block in the editor, it will be used to search for help articles based on the block name.
```

Comment:
Let's move this comment to the first line of the function. This makes more sense for explaining the const `blockSearchQuery`.

We can add a new comment to explain what `useContextBasedSearchingMapping` is doing.

Here is an example you can modify:

```
Returns the search query and tailored articles based on the current route.
The search query will be determined based on the selected Gutenberg block, an exact match with URL mapping, or a fuzzy match (in this specific order).
The tailored Articles will be determined based on an exact URL match within the tailored article mapping.
```